runtime/sema/check_composite_declaration.go

```diff
@@ -405,9 +497,26 @@ func (checker *Checker) declareNestedDeclarations(
 		nestedCompositeTypes = append(nestedCompositeTypes, nestedCompositeType)
 	}
 
+	// Declare nested attachments
+
+	for _, nestedDeclaration := range nestedAttachmentDeclaration {
+		if _, exists := nestedDeclarations[nestedDeclaration.Identifier.Identifier]; !exists {
+			nestedDeclarations[nestedDeclaration.Identifier.Identifier] = nestedDeclaration
+		}
```

Comment:
Maybe factor out the repeated `nestedDeclaration.Identifier.Identifier` into variable